src/plugins/discover/public/application/view_components/canvas/top_nav.tsx

```diff
@@ -34,7 +34,7 @@ export interface TopNavProps {
 
 export const TopNav = ({ opts, showSaveQuery, isEnhancementsEnabled }: TopNavProps) => {
   const { services } = useOpenSearchDashboards<DiscoverViewServices>();
-  const { inspectorAdapters, savedSearch, indexPattern } = useDiscoverContext();
+  const { data$, inspectorAdapters, savedSearch, indexPattern } = useDiscoverContext();
```

Comment:
if we only require do we think we should maybe just create a new observable that this listens to instead of subscribing to a lot of data in the top nav?